ntbs-service/Services/LegacySearchBuilder.cs

```diff
@@ -19,27 +19,58 @@ public class LegacySearchBuilder : ILegacySearchBuilder
         
         public LegacySearchBuilder()
         {
-            this.sqlQuery = "";
             this.parameters = new ExpandoObject();
         }
 
         public ISearchBuilder FilterById(string id)
         {
             if (!string.IsNullOrEmpty(id))
             {
-                sqlQuery += "WHERE n.OldNotificationId = @id OR n.GroupId = @id AND n.Source = 'LTBR' OR dmg.NhsNumber = @id";
+                sqlQuery += @"WHERE dmg.OldNotificationId = @id OR n.GroupId = @id AND n.Source = 'LTBR' OR dmg.NhsNumber = @id
+                    ";
                 parameters.id = id;
             }
             return this;
         }
 
         public ISearchBuilder FilterByFamilyName(string familyName)
         {
+            if (!string.IsNullOrEmpty(familyName))
+            {
+                var wildcardedFamilyName = '%' + familyName + '%';
+                if(sqlQuery == null)
+                {
+                    sqlQuery += @"WHERE dmg.FamilyName LIKE @familyName
+                        ";
+                }
+                else
+                {
+                    sqlQuery += @"AND dmg.FamilyName LIKE @familyName
+                        ";
+                }
+                
+                parameters.familyName = wildcardedFamilyName;

```

Comment:
There's a bunch of repetition in the form of:
```
    if(sqlQuery == null)
                {
                    sqlQuery += @"WHERE <condition>
                        ";
                }
                else
                {
                    sqlQuery += @"AND <condition>
                        ";
                }
    }
```
perhaps that's something we can either capture in a common method (`appendCondition(<condition>`), or perhaps we build up a collection of conditions and join them correctly in the `GetResult` method?